cmd/remoterun/ssh.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package remoterun
+
+import (
+	"bufio"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strings"
+
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/remote"
+)
+
+type sshForwarder struct {
+	getTLSConfig func() *tls.Config
+}
+
+func newSSHForwarder() *sshForwarder {
+	return &sshForwarder{
+		getTLSConfig: newTLSConfigWithSystemCA,
+	}
+}
+
+// CA is not specified because it should use System CA root. When remote-run command is executed,
+// the CA should be already available as the CLI set it in the Dockerfile used as a base to
+// run the command
+func newTLSConfigWithSystemCA() *tls.Config {
+	return &tls.Config{}
+}
+
+// startSshForwarder This functions starts the ssh-forwarded process expected to be run on remote-run commands.
+// This process just listens a UNIX socket, and forward the messages to the SSH Agent exposed in the hostname
+// and port received as parameters
+func (s *sshForwarder) startSshForwarder(sshAgentHostname, sshAgentPort, userToken string) {
+	// Remove existing socket if it exists
+	os.Remove(remote.SshAgentLocalSocket)
+
+	// Listen on the UNIX domain socket
+	localListener, err := net.Listen("unix", remote.SshAgentLocalSocket)
+	if err != nil {
+		oktetoLog.Fatalf("Failed to listen on UNIX socket %s: %v", remote.SshAgentLocalSocket, err)
```

Comment:
Do we need to exit the full program? (this also affects the main execution. Here is a playground [showing this](https://go.dev/play/p/bQ9qDsXDiIp)). Can we consider returning a proper error to the user with our format?

Reply:
Good catch! I did it in that way to catch faster issue while implementing it gut forgot to change it. It returns an error now, and it doesn't panic